### ISSUE TITLE
start time is a tuple, not a string.

### ISSUE THREAD

**Issue by andreas-ibm** [coreemu/core#179]
This is one of the ones where you wonder how it ever worked...!

Setting start time on a service (so underlying stuff has time to get initialized), boom.

what appears to happen is in `service.py` in `setvalue(self, key, value)`

```python
        if value:
            if key == "startidx":
                value = int(value)
            elif key == "meta":
                value = str(value)
            else:
                value = utils.make_tuple_fromstr(value, str)

        if key == "dirs":
            self._dirs = value
        elif key == "files":
            self._configs = value
        elif key == "startidx":
            self._startindex = value
        elif key == "cmdup":
            self._startup = value
        elif key == "cmddown":
            self._shutdown = value
        elif key == "cmdval":
            self._validate = value
        elif key == "meta":
            self._meta = value
        elif key == "starttime":
            self._starttime = value
```
So start time, it a tuple because it's not startindex or meta...

then in `bootnodeservices(self, node)` we get
```python
            if len(str(s._starttime)) > 0:
                try:
                    logger.debug(type(s._starttime))
                    logger.debug(s._starttime)
                    t = float(s._starttime)
                    if t > 0.0:
                        fn = self.bootnodeservice
                        self.session.event_loop.add_event(t, fn, node, s, services, False)
                        continue
                except ValueError:
                    logger.exception("error converting start time to float")
            self.bootnodeservice(node, s, services, use_startup_service)
```

Well that's never going to work, because we explicitly set it to a tuple!

So have I missed something, or should I just submit a PR to add starttime to the list of things to just treat as a string?

**Comment by bharnden** [comment]
Added fix in rel/5.1, need to determine how to treat it, if at all for rel/5.2

**Comment by andreas-ibm** [comment]
As a parting comment; the dependencies might not fully cut it. For example on my DTN service, it fails to start up if it's started too quickly after the network interfaces have been created as it fails to attach properly to the interface, so I specified a couple seconds delay in the start time and it starts fine....
